### ISSUE TITLE
Missing pixel when resizing due to math.Floor

### ISSUE THREAD

**Issue by rogersole** [jcupitt/libvips#509]
When trying to resize an image of 1920x1080 to a fixed width to 300 (with no height specification), the resulting image has a width of 299 pixels.

Digging into the vips source code, I found that changing the `math.Floor` to `math.Ceil` prevents this rounding down that's always performed, causing the "missing pixel" issue:

In `resize.go`:

```
func imageCalculations(o *Options, inWidth, inHeight int) float64 {
	factor := 1.0
	xfactor := float64(inWidth) / float64(o.Width)
	yfactor := float64(inHeight) / float64(o.Height)

	switch {
	// Fixed width and height
	case o.Width > 0 && o.Height > 0:
		if o.Crop {
			factor = math.Min(xfactor, yfactor)
		} else {
			factor = math.Max(xfactor, yfactor)
		}
	// Fixed width, auto height
	case o.Width > 0:
		factor = xfactor
		o.Height = int(math.Ceil(float64(inHeight) / factor))
	// Fixed height, auto width
	case o.Height > 0:
		factor = yfactor
		o.Width = int(math.Ceil(float64(inWidth) / factor))
	// Identity transform
	default:
		o.Width = inWidth
		o.Height = inHeight
		break
	}

	return factor
}
```

Is it a `math.Floor` for any specific purpose?

**Comment by jcupitt** [comment]
Hi @rogersole, have a look at #491, it has some background. There's a branch, which will probably get merged for 8.4, that switches to `round()`.

In the short term, add 0.1 to the target size when calculating the scale factor to prevent unexpected round down due to rounding errors. Longer term, this hack will probably become unnecessary (but not harmful).

**Comment by jcupitt** [comment]
... I'll close, further discussion should be in #491.
